src/main/java/com/google/sps/servlets/CalendarServlet.java

```diff
@@ -51,6 +51,14 @@
   protected void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException {
 
+    UserService userService = UserServiceFactory.getUserService();
+    if(!userService.isUserLoggedIn()) {
+        response.setContentType("application/text");
+        response.getWriter().println("You must be logged in to use the calendar function");
+        response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+        return;
+    }
```

Comment:
This is something that would potentially be used in other servlets as well. So maybe move this logic into a common Util class so that you don't need to duplicate the code across servlets?